libs/Microsoft.MixedReality.WebRTC.Native/src/interop/audio_transceiver_interop.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// This is a precompiled header, it must be on its own, followed by a blank
+// line, to prevent clang-format from reordering it with other headers.
+#include "pch.h"
+
+#include "interop/audio_transceiver_interop.h"
+#include "media/audio_transceiver.h"
+
+using namespace Microsoft::MixedReality::WebRTC;
+
+void MRS_CALL
+mrsAudioTransceiverAddRef(mrsAudioTransceiverHandle handle) noexcept {
+  if (auto transceiver = static_cast<AudioTransceiver*>(handle)) {
+    transceiver->AddRef();
+  } else {
+    RTC_LOG(LS_WARNING)
+        << "Trying to add reference to NULL AudioTransceiver object.";
+  }
+}
+
+void MRS_CALL
+mrsAudioTransceiverRemoveRef(mrsAudioTransceiverHandle handle) noexcept {
+  if (auto transceiver = static_cast<AudioTransceiver*>(handle)) {
+    transceiver->RemoveRef();
+  } else {
+    RTC_LOG(LS_WARNING)
+        << "Trying to remove reference from NULL AudioTransceiver object.";
+  }
+}
+
+void MRS_CALL mrsAudioTransceiverRegisterStateUpdatedCallback(
+    mrsAudioTransceiverHandle handle,
+    mrsAudioTransceiverStateUpdatedCallback callback,
+    void* user_data) noexcept {
+  if (auto transceiver = static_cast<AudioTransceiver*>(handle)) {
+    transceiver->RegisterStateUpdatedCallback(
+        Transceiver::StateUpdatedCallback{callback, user_data});
+  }

```

Comment:
What's the criteria for warning on null handles or not?

Reply:
It is very rare that an API call accepts NULL handles. Generally those are invalid. Only one that comes to mind is changing the transceiver's local sender track to NULL to detach the previous one. So most of the time we emit warnings or return an error `kInvalidNativeHandle`.